core/src/test/java/org/elasticsearch/action/admin/indices/upgrade/UpgradeReallyOldIndexTest.java

```diff
@@ -38,11 +40,25 @@ public void testUpgrade_0_90_6() throws Exception {
         assertTrue(UpgradeTest.hasAncientSegments(client(), indexName));
         assertNoFailures(client().admin().indices().prepareUpgrade(indexName).setUpgradeOnlyAncientSegments(true).get());
 
-        assertFalse(UpgradeTest.hasAncientSegments(client(), "index-0.90.6"));
+        assertFalse(UpgradeTest.hasAncientSegments(client(), indexName));
         // This index has only ancient segments, so it should now be fully upgraded:
         UpgradeTest.assertUpgraded(client(), indexName);
         assertEquals(Version.CURRENT.luceneVersion.toString(), client().admin().indices().prepareGetSettings(indexName).get().getSetting(indexName, IndexMetaData.SETTING_VERSION_MINIMUM_COMPATIBLE));
         assertMinVersion(indexName, Version.CURRENT.luceneVersion);
+
+        assertEquals(client().admin().indices().prepareGetSettings(indexName).get().getSetting(indexName, IndexMetaData.SETTING_VERSION_UPGRADED), Integer.toString(Version.CURRENT.id));
+    }
+
+    public void testUpgradeConflictingMapping() throws Exception {
+        String indexName = "index-1.7.0";
```

Comment:
Can we name this index differently to indicate it is an error case? Perhaps index-conflicting-mappings-1.7.0?